owdex/linkmanager.py

```diff
@@ -1,95 +1,147 @@
+import re
+from uuid import uuid4 as uuid
+
 import flask as f
 from flask import current_app as app
 
-import pysolr
 from bs4 import BeautifulSoup as bs
+from pysolr import Solr
 from requests import get
 from url_normalize import url_normalize
 
 
+class Link:
+    def __init__(
+        self,
+        *,
+        url,
+        title,
+        submitter,
+        index=None,
+        id=None,
+        content=None,
+        description=None,
+        score=0,
+        **kwargs,
+    ):
+        self.url = url_normalize(url)
+        self.title = title
+        self.index = index
+        self.submitter = submitter
+        self.score = score
+        self.__dict__.update(kwargs)
+
+        if content and description and id:
+            self.content, self.description, self.id = content, description, id
+        else:
+            self.id = str(uuid())
+            self.content, self.description = self.scrape()
+
+    def scrape(self):
+        response = get(self.url).text
```

Comment:
Possibly adding a new method `scrape` to the linkmanager

Reply:
You're right with the `@dataclass`. Owdex is the first real project I've done in a while and so I'm not fully up to date with certain Python 3.latest features. That's definitely a useful one.

Why move `scrape` to `LinkManager`? Doesn't it work fine on `Link`?

I feel like this is a pretty challenging thing to implement in OOP in a way that makes a huge amount of sense, because `Link` objects aren't used much -- the database stores JSON and the user sees HTML. Maybe it would be wiser to remove the `Link` abstraction layer.

Reply:
Resolved in [774c0fd](https://github.com/owdex/owdex/pull/72/commits/774c0fdc0e9d26caa68901af33409fe4bfeef9da).